internal/manifests/mutate.go

```diff
@@ -12,10 +13,22 @@ import (
 	corev1 "k8s.io/api/core/v1"
 	networkingv1 "k8s.io/api/networking/v1"
 	rbacv1 "k8s.io/api/rbac/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+// ImmutableErr occurs if an immutable field should be changed.
+type ImmutableErr struct {
+	field    string
+	existing interface{}
+	desired  interface{}
+}
+
+func (m *ImmutableErr) Error() string {
+	return fmt.Sprintf("update to immutable field %s is forbidden", m.field)
```

Comment:
shouldn't it print the existing and desired as well?

Reply:
I did initially, but printing structs with `fmt.Sprintf("%v", some_struct)` is an unreadable mess if the struct is big.

Reply:
shall we then remove fields that are not used in the error struct?

Reply:
I updated the message to show the result of `cmp.Diff()` now - it's still a bit unreadable as it's a single line, but when replacing the `\n` and `\t` we see a nice diff in the logs, and we already had a dependency on this library anyway (https://github.com/google/go-cmp).